iothub_client/src/iothub_client_core_ll.c

```diff
@@ -399,7 +398,44 @@ static int create_blob_upload_module(IOTHUB_CLIENT_CORE_LL_HANDLE_DATA* handle_d
     }
     else
     {
-        result = 0;
+        IOTHUB_CREDENTIAL_TYPE credential_type = IoTHubClient_Auth_Get_Credential_Type(handle_data->authorization_module);
+
+        if (use_dev_auth &&
```

Comment:
Are there other types of credentials we'd want to propagate as well?

Reply:
Yes, there is DEVICE KEYs and SAS, but these are discouraged methods when using DPS. Customers should use x509 with DPS. Thus we are not prioritizing fixing the path for other auth methods (with DPS! When using IoT Hub directly users can still use any auth mode they desire to use upload to blob).